src/app/modules/core/services/beacon-node.service.ts

```diff
@@ -25,16 +36,41 @@ export class BeaconNodeService {
   ) { }
 
   private apiUrl = this.environmenter.env.validatorEndpoint;
-  private beaconNodeEndpoint: string;
-
-  // Observables.
-  status$ = this.http.get<NodeConnectionResponse>(`${this.apiUrl}/health/node_connection`).pipe(
-    tap((res: NodeConnectionResponse) => {
-      this.beaconNodeEndpoint = res.beaconNodeEndpoint + BEACON_API_SUFFIX;
-    }),
-  );
+  beaconState: BeaconState = new BeaconState();
+
+  private get getBeaconEndpoint$(): BehaviorSubject<string> {
+    if (!this.beaconState.beaconNodeEndpoint$.getValue()) {
+      this.http.get<NodeConnectionResponse>(`${this.apiUrl}/health/node_connection`).subscribe(
```

Comment:
Let's try to avoid any logic in the subscribe functions in the code and instead proceed with pure, functional-reactive methods in the services